test/unit/permissions.test.ts

```diff
@@ -0,0 +1,62 @@
+import { get, merge } from "lodash";
+import { pluginConfigExt, runServerless } from "../utils/runServerless";
+
+describe("permissions", () => {
+    it("should not override user-defined role", async () => {
+        const { cfTemplate } = await runServerless({
+            fixture: "permissions",
+            configExt: merge(pluginConfigExt, {
+                provider: {
+                    iam: {
+                        role: "arn:aws:iam::123456789012:role/role",
+                    },
+                },
+            }),
+            cliArgs: ["package"],
+        });
+        expect(cfTemplate.Resources.FooLambdaFunction).toMatchObject({
+            Properties: {
+                Role: "arn:aws:iam::123456789012:role/role",
+            },
+        });
+    });
+
+    it("should append permissions when using the deprecated iamRoleStatements", async () => {
+        const { cfTemplate } = await runServerless({
+            fixture: "permissions",
+            configExt: merge(pluginConfigExt, {
+                provider: {
+                    iam: {
+                        role: {
+                            statements: [
```

Comment:
The test name mentions `iamRoleStatements` but actually uses `iam.role.statements`, is this a mistake?

Reply:
yes it is :sweat_smile: